### PR TITLE
Mediatr example

### PR DESCRIPTION
**Pull request by efleming18**

### CODE REVIEW

**Comment by efleming18** [comment]
Providing an example for issue: https://github.com/dotnet-architecture/eShopOnWeb/issues/239

File: src/Web/Features/Orders/GetMyOrders.cs [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+using System.Linq;
+using MediatR;
+using Microsoft.eShopWeb.ApplicationCore.Interfaces;
+using Microsoft.eShopWeb.ApplicationCore.Specifications;
+using Microsoft.eShopWeb.Web.ViewModels;
+using System.Collections.Generic;
+
+namespace Microsoft.eShopWeb.Web.Features.Orders
+{
+    public class GetMyOrders : IRequest<IEnumerable<OrderViewModel>>
```
Comment (ardalis) [inline src/Web/Features/Orders/GetMyOrders.cs]: I'd probably keep to one class per file, unless you're using inner classes, which I'm not sure makes sense here. Your folder/namespace organization here should keep them together well enough, I think.

File: src/Web/Features/Orders/GetMyOrders.cs [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+            UserName = userName;
+        }
+    }
+
+    public class GetMyOrdersHandler : IRequestHandler<GetMyOrders, IEnumerable<OrderViewModel>>
+    {
+        private readonly IOrderRepository _orderRepository;
+
+        private readonly string _defaultOrderStatus = "Pending";
+        private readonly decimal _defaultOrderDiscount = 0;
```
Comment (ardalis) [inline src/Web/Features/Orders/GetMyOrders.cs]: Should these just be set as the default values on `OrderViewModel`?

File: src/Web/Features/Orders/RequestHandlers/GetMyOrdersHandler.cs [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+using Microsoft.eShopWeb.ApplicationCore.Interfaces;
+using Microsoft.eShopWeb.ApplicationCore.Specifications;
+using Microsoft.eShopWeb.Web.Features.Orders.Requests;
+using Microsoft.eShopWeb.Web.ViewModels;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.eShopWeb.Web.Features.Orders.RequestHandlers
```
Comment (ardalis) [inline src/Web/Features/Orders/RequestHandlers/GetMyOrdersHandler.cs]: Probably better to put each endpoint's things in a folder, right?
```
/MyOrders/GetMyOrdersRequest
/MyOrders/GetMyOrdersHandler
/OrderDetails/GetOrderDetailsRequest
/OrderDetails/GetOrderDetailsHandler
```

File: src/Web/ViewModels/OrderViewModel.cs [hunk 4]
```diff
@@ -9,12 +9,8 @@ public class OrderViewModel
         public int OrderNumber { get; set; }
         public DateTimeOffset OrderDate { get; set; }
         public decimal Total { get; set; }
-        public string Status { get; set; }
-
-        public Address ShippingAddress { get; set; } 
-
+        public string Status => "Pending";
```
Comment (ardalis) [inline src/Web/ViewModels/OrderViewModel.cs]: Magic string. Use a constant?

**Comment by efleming18** [comment]
Closing this PR for now until we're ready to merge it in.
Will leave this branch out there though as an example, and when/if we want to merge it in, it will be ready.

**Pull request closed without merging**